src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String model;
+    private int number;
+    private Engine engine;
+    private String color;
+    private int gasolineVolume;
 
+    public Car(String model, int number, Engine engine, String color, int gasolineVolume) {
+        this.model = model;
+        this.number = number;
+        this.engine = engine;
+        this.color = color;
+        this.gasolineVolume = gasolineVolume;
+    }
+
+    public Engine getEngine() {
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() {
+        try {
+            Car clonedCar = (Car) super.clone();
+            clonedCar.setEngine(engine.clone());
+            return clonedCar;
```

Comment:
you can inline `clone()` implementation using constructor

Reply:
done